### ISSUE TITLE
Support Django's `contenttypes` Framework

### ISSUE THREAD

**Issue by rsomani95** [MrThearMan/graphene-django-query-optimizer#66]
### I have read the docs thoroughly before making this feature request.

- [X] Yes

### I have read through other open issues, and my issue is not a duplicate.

- [X] Yes

### Description

I believe there's a 2N+1 bug when working with Django's [`contenttypes` framework](https://docs.djangoproject.com/en/5.0/ref/contrib/contenttypes/#generic-relations). I've done some digging and documented my findings below. First, I've shown the bug using the `taggit` library, which relies on the aforementioned framework, and then re-produced a stripped down barebones version to confirm that the issue is indeed the incompatibility with Django's offering, and not the `taggit` library.

# Real World Example -- Working With The `taggit` Library

```python
from django.db import models
from django.utils.translation import pgettext_lazy
from taggit.models import GenericTaggedItemBase, TagBase

# The name `TaggableManager` is misleading, as this is actually a custom many to many `RelatedField`
from taggit.managers import TaggableManager as TaggableManyToManyRelatedField


class Tag(TagBase):
    category = models.CharField(db_index=True, default="custom", max_length=200, null=False)
    name = models.CharField(
        db_index=True, max_length=255, verbose_name=pgettext_lazy("A tag name", "name")
    )


class TaggedItem(GenericTaggedItemBase):
    class Meta:
        unique_together = [["content_type", "object_id", "tag"]]

    tag = models.ForeignKey(Tag, related_name="custom_tagged_items", on_delete=models.CASCADE)
    confidence = models.FloatField()


class PropertyManager(models.Model):
    name = models.CharField(max_length=1024)
    email = models.EmailField(blank=True)

    tags = TaggableManyToManyRelatedField(blank=True, through=TaggedItem)
```

`TaggableManyToManyRelatedField` is, as I understand it, a [custom many to many field](https://github.com/jazzband/django-taggit/blob/5dded2cb8935cb8118dbd8595602d708bd2e8ae2/taggit/managers.py#L431-L453), and the `TaggedItem` model relies on the Django [`contenttypes`](https://docs.djangoproject.com/en/5.0/ref/contrib/contenttypes/) framework to work with generic foreign keys

On the GQL side, I've wired things up as follows:

```python
from query_optimizer import DjangoObjectType

class TaggedItemType(DjangoObjectType):
    class Meta:
        model = TaggedItem
        fields = ["confidence"]

    name = graphene.String()
    category = graphene.String()

    @required_fields("tag__name")
    def resolve_name(model: TaggedItem, info: GQLInfo):
        return model.tag.name

    @required_fields("tag__category")
    def resolve_category(model: TaggedItem, info: GQLInfo):
        return model.tag.category


class HousingCompanyType(DjangoObjectType):
    ...
    tagged_items = DjangoListField(TaggedItemType)


# NOTE: This is being done to prevent graphene from complaining. For the moment, I'm
# not querying `tags` directly, but defining a custom `tagged_items = DjangoListField(TaggedItemType)`
# We can ignore this, but will need it to get the schema to compile
# Doing this "properly" warrants a separate issue perhaps so I'm avoiding it for now
@convert_django_field.register(TaggableManyToManyRelatedField)
def convert_field_to_string(field, registry=None):
    return graphene.String(description=field.help_text, required=not field.null)
```

---

Now, when I query as follows:

```graphql
{
  allHousingCompanies {
    name
    taggedItems {
      confidence
    }
  }
}
```

It makes 2N queries (where N is the number of `tagged_items`, not the objects... and an object can have multiple tags, but only has 1 in my test data).

<details>
<summary>SQL generated</summary>

```sql
SELECT "example_housingcompany"."id", "example_housingcompany"."name" FROM "example_housingcompany" ORDER BY "example_housingcompany"."name" ASC
SELECT "example_taggeditem"."id", "example_taggeditem"."confidence" FROM "example_taggeditem" WHERE ("example_taggeditem"."content_type_id" = %s AND "example_taggeditem"."object_id" IN (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, 
%s, %s, %s, %s, %s, %s, %s, %s, %s, %s))
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
```

</details>

Clearly, it's making additional `"object_id"` and `"content_type_id"` queries for _each_ tagged item.
I was able to mitigate this by exposing the `"object_id"` and querying for it, like so:

```python
class TaggedItemType(DjangoObjectType):
    class Meta:
        model = TaggedItem
        fields = ["confidence", "object_id"]  # Added 'object_id' here

    ...
    # The rest remains the same as earlier
```

```graphql
{
  allHousingCompanies {
    name
    taggedItems {
      confidence
      objectId
    }
  }
}
```

<details>
<summary>SQL generated</summary>

```sql
SELECT "example_housingcompany"."id", "example_housingcompany"."name" FROM "example_housingcompany" ORDER BY "example_housingcompany"."name" ASC
SELECT "example_taggeditem"."id", "example_taggeditem"."object_id", "example_taggeditem"."confidence" FROM "example_taggeditem" WHERE ("example_taggeditem"."content_type_id" = %s AND "example_taggeditem"."object_id" IN (%s, %s,
%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s))
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
SELECT "example_taggeditem"."id", "example_taggeditem"."content_type_id" FROM "example_taggeditem" WHERE "example_taggeditem"."id" = %s LIMIT 21
```

</details>

However, I wasn't able to do the same with `"content_type"` and `"content_type_id"`. I suspect this has to do because of nuances in Django's internals that are currently beyond me.

Additionally, if I query for fields of the parent `Tag` object, which were decorated with `required_fields`:

```graphql
{
  allHousingCompanies {
    name
    taggedItems {
      name
      category
      confidence
    }
  }
}
```

This causes another N queries to occur.

---

## Stripped Down Example

To isolate the issue, I created a simpler model that doesn't have the `taggit` dependency, but uses Django's `contenttypes` framework to achieve the generic foreign key interface. This model is almost identical to the one shown in [Django's docs](https://docs.djangoproject.com/en/5.0/ref/contrib/contenttypes/#generic-relations).

Model definition:
```python
from django.db import models
from django.contrib.contenttypes.fields import GenericForeignKey
from django.contrib.contenttypes.models import ContentType
from django.utils.translation import gettext_lazy as _
from django.utils.translation import pgettext_lazy


class CustomTag(models.Model):
    # Generic foreign key setup via Django's ContentType mechanism
    content_object = GenericForeignKey("content_type", "object_id")
    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
    object_id = models.IntegerField(verbose_name=_("object ID"), db_index=True)

    # Tag stuff
    category = models.CharField(db_index=True, default="custom", max_length=200, null=False)
    name = models.CharField(
        db_index=True, max_length=255, verbose_name=pgettext_lazy("A tag name", "name")
    )
    confidence = models.FloatField()

class HousingCompany(models.Model):
    ...
    custom_tags = GenericRelation(
        CustomTag, object_id_field="object_id", content_type_field="content_type"
    )
```

One thing worth noting here is that adding / omitting the `GenericRelation` here makes no difference to the [generated migration](https://github.com/rsomani95/graphene-django-query-optimizer/blob/c9e3ef89ab73f0f5c7d8169663aab066a0a6b013/tests/example/migrations/0003_customtag.py#L44-L64), despite the docs saying the relationship is not inferred directly.

GQL Wiring:
```python
from query_optimizer import DjangoObjectType

class CustomTagType(DjangoObjectType):
    class Meta:
        model = CustomTag
        fields = ["confidence", "name", "category"]

class HousingCompanyType(DjangoObjectType):
    ...
    custom_tags = DjangoListField(CustomTagType)
```

GQL Query:
```graphql
{
  allHousingCompanies {
    name
    customTags {
      name
      category
      confidence
    }
  }
}
```

This again generates 2N queries.

<details>
<summary>SQL generated</summary>

```sql
SELECT "example_housingcompany"."id", "example_housingcompany"."name" FROM "example_housingcompany" ORDER BY "example_housingcompany"."name" ASC
SELECT "example_customtag"."id", "example_customtag"."category", "example_customtag"."name", "example_customtag"."confidence" FROM "example_customtag" WHERE ("example_customtag"."content_type_id" 
= %s AND "example_customtag"."object_id" IN (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s))
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."object_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
SELECT "example_customtag"."id", "example_customtag"."content_type_id" FROM "example_customtag" WHERE "example_customtag"."id" = %s LIMIT 21
```

</details>

Not surprisingly, doing the exercise above with exposing and querying for `object_id` mitigates the issue down to N queries only.

---

## Potential Solution

An obvious step to try seemed like directly joining the `"content_type"` table to the `CustomTag` queryset, which I tried as follows:
```python
class CustomTagType(DjangoObjectType):
    class Meta:
        model = CustomTag
        fields = ["confidence", "name", "category", "object_id"]

    @classmethod
    def get_queryset(cls, queryset: QuerySet[CustomTag], info: GQLInfo) -> QuerySet[CustomTag]:
        queryset = queryset.select_related("content_type")
        return super().get_queryset(queryset, info)
```

This didn't work while querying in GQL because the optimizer is doing its work modifying the queryset. I failed with the following error:

```
"Field CustomTag.content_type cannot be both deferred and traversed using select_related at the same time."
```

I'm not sure where to go from here, and would love to hear your thoughts.

---

### To Reproduce

For ease of reproduction, I've committed all of the above investigations to a branch off the `0.2.0` release here: https://github.com/rsomani95/graphene-django-query-optimizer/tree/repr

### Motivation

Use the [`django-taggit`](https://github.com/jazzband/django-taggit) library optimally with the optimizer to enable a flexible tagging interface in my project

### Would you like to solve this issue yourself with a pull request?

Yes

**Comment by rsomani95** [comment]
I have a potential WIP solution in https://github.com/MrThearMan/graphene-django-query-optimizer/pull/67

**Comment by MrThearMan** [comment]
Seem to me like this is a more general issue about supporting `GenericForeignKey`/`GenericRelation`. I see you already opened a pull request, so I added my comments there. I assume you will be working on a permanent solution, so I'll assign this issue to you for now. I don't currrently have a use for this myself, so take your time to think it through, but if you need help, just ask.

**Comment by rsomani95** [comment]
@MrThearMan thanks a lot for your feedback. All your comments on the PR make sense. Yes, I'd definitely like to work towards a permanent solution. I will step towards it as I need to take some shortcuts in the short term to get a functional integration up and running in my work project.

**Comment by MrThearMan** [comment]
Hi, I had some time on my hands, so I decided to see if this could be implemented. Managed to add support for both `GenericForeignKey` and `GenericRelation`, although `GenericForeignKey` would require [GenericPrefetch](https://docs.djangoproject.com/en/5.0/ref/contrib/contenttypes/#genericprefetch) from Django 5.0 to be able to optimize further relations. I'm not willing to drop support for 4.2 yet, as it is the latest LTS, and 5.0 is quite new still.

Released in [0.5.1](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.5.1).
